### PR TITLE
Add jitter to sleep

### PR DESCRIPTION
**Pull request by ashi009**
Add `sleep-jitter` config/flag to sleep. So when deployed in large scale, this will reduce the likelihood for
having all the servers banning the remote endpoint at once.

In this change `runChan` is replaced with a lambda function `runOnce`, which is a non-blocking function
to trigger a single run, and it returns if ansible is running already. Which removes the possibility that multiple
ansible runs queued up and runs back to back.

When sleep jitter is specified, the puller will sleep a duration that conforms to uniform distribution 
`U(sleep-jitter, sleep+jitter)`. Otherwise the behavior remains unchanged.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -296,24 +298,39 @@ func main() {
 	promVersion.WithLabelValues(Version).Set(1)
 
 	period := time.Duration(viper.GetInt("sleep")) * time.Minute
-	tickerChan := time.NewTicker(period).C
-	runChan := make(chan bool, 8)
+	jitter := time.Duration(viper.GetInt("sleep-jitter")) * time.Minute
+
+	if jitter >= period {
+		logrus.Fatalln("sleep-jitter is too large")
```
Comment (jyates) [inline main.go:304]:
```suggestion
logrus.Fatalln("sleep-jitter is too large, it must be less than the 'sleep' period %s", sleep)
```

File: main.go [hunk 2]
```diff
@@ -296,24 +298,39 @@ func main() {
...
+			for range time.Tick(period) {
+				runOnce()
+			}
+			return
+		}
+		rng := rand.New(rand.NewSource(time.Now().Unix()))
 		for {
-			<-tickerChan
-			runChan <- true
+			time.Sleep(period - jitter + time.Duration(rng.Intn(2*int(jitter))))
```
Comment (jyates) [inline main.go:326]: This is a subtle way to calculate this - worth a comment

**Review by jyates — APPROVED** [review]
Would be good to have @shk3 take a pass over it too

File: main.go (hunk 2 above)
Comment (shk3) [inline main.go:327]: Int63n? https://pkg.go.dev/time#Duration

**Review by shk3 — APPROVED** [review]

File: main.go [hunk 3]
```diff
@@ -296,24 +298,39 @@ func main() {
 	promVersion.WithLabelValues(Version).Set(1)
 
 	period := time.Duration(viper.GetInt("sleep")) * time.Minute
-	tickerChan := time.NewTicker(period).C
-	runChan := make(chan bool, 8)
+	jitter := time.Duration(viper.GetInt("sleep-jitter")) * time.Minute
+
+	if jitter >= period {
```
[the remark this answers is not in the document]
Comment (ashi009) [reply main.go:303]: don't bother? sleep is not checked either.

**Review by shk3 — APPROVED** [review]

**Review by shk3 — APPROVED** [review]

**Pull request merged**